### PR TITLE
[Sival, i2c] Adds Clock stretching test

### PR DESCRIPTION
**Pull request by engdoreis**
Fix https://github.com/lowRISC/opentitan/issues/19838

### CODE REVIEW

File: sw/device/lib/testing/json/command.h [hunk 1]
```diff
@@ -21,6 +21,7 @@ extern "C" {
     value(_, I2cStartTransferWriteSlow) \
     value(_, I2cStartTransferRead) \
     value(_, I2cStartTransferWriteRead) \
+    value(_, I2cTestConfiging) \
```
Comment (jwnrt) [inline sw/device/lib/testing/json/command.h:24]:
```suggestion
    value(_, I2cTestConfig) \
```

File: sw/host/opentitanlib/src/test_utils/i2c_target.rs [hunk 2]
```diff
@@ -74,3 +74,12 @@ impl I2cTransferStart {
         Self::recv(uart, Duration::from_secs(300), false)
     }
 }
+
+impl I2cTestConfig {
+    pub fn write(&self, uart: &dyn Uart) -> Result<()> {
+        TestCommand::I2cTestConfiging.send(uart)?;
```
Comment (jwnrt) [inline sw/host/opentitanlib/src/test_utils/i2c_target.rs:80]:
```suggestion
        TestCommand::I2cTestConfig.send(uart)?;
```

File: sw/device/tests/i2c_target_test.c [hunk 3]
```diff
@@ -307,6 +309,13 @@ static status_t command_processor(ujson_t *uj) {
         // at 100 KHz, forcing the peripheral to stretch the clock.
         RESP_ERR(uj, start_write_transaction(uj, &i2c, kTransactionDelay));
         break;
+      case kTestCommandI2cTestConfiging: {
```
Comment (jwnrt) [inline sw/device/tests/i2c_target_test.c:312]:
```suggestion
      case kTestCommandI2cTestConfig: {
```

**Review by jwnrt — APPROVED** [review]
Mostly LGTM, though I'd suggest `Config` rather than `Configing`

File: sw/device/lib/testing/json/command.h (hunk 1 above)
Comment (engdoreis) [reply sw/device/lib/testing/json/command.h]: Good catch, that was a failed search and replace.

**Comment by engdoreis** [comment]
CI passed all the tests, but timed out during the results publishing.

**Comment by jwnrt** [comment]
I forgot to increase the timeout on the `earlgrey_es_sival` branch. See https://github.com/lowRISC/opentitan/pull/22774 for a fix.

**Pull request merged**